eodatasets3/wagl.py

```diff
@@ -121,10 +133,147 @@ def _unpack_products(
                     )
 
 
+def missing_packets(dataset: h5py.Dataset, granule):
+    """
+    If missing packets, returns metadata dict and None if not
+    """
+    # Define bands to be masked
+    band_ids_mask = {str(i) for i in range(0, 13)}
+    band_ids_mask.add("8A")
+    if dataset.attrs["band_id"] not in band_ids_mask:
+        return None
+
+    platform_name = granule.wagl_metadata["source_datasets"]["platform_id"].lower()
+    if "sentinel" not in platform_name:
+        return None
+
+    l1_zip_path = Path(granule.wagl_metadata["source_datasets"]["source_level1"])
+    xml_file_path = Path(
+        granule.source_level1_metadata.accessories["metadata:s2_tile"].path
+    )
+
+    # Parse tile metadata xml file
+    with zipfile.ZipFile(l1_zip_path, "r") as z:
+        return missing_packets_from_xml(z.read(xml_file_path.as_posix()))
+
+
+def missing_packets_from_xml(xml: bytes):
+    root = Et.fromstring(xml, forbid_dtd=True)
+
+    try:
+        msi_data_percentage = float(root[2][0][1].text)
+    except IndexError:
+        return None
+
+    if msi_data_percentage <= 0.0:
+        return None
+
+    children = root[2][1]
```

Comment:
Similar fragility (depends too much on document structure).

See my code snipped below for alternative